File: backend/python/test_db.py
"""
Test database connection for Python backend
"""
import sys
sys.path.append('.')

from database import test_connection, fetch_all, fetch_one

def main():
    print("🔍 Testing PostgreSQL connection...\n")
    
    if not test_connection():
        print("\n❌ Failed to connect to database")
        return
    
    # Check routes
    routes = fetch_all("SELECT COUNT(*) as count FROM routes")
    print(f"\n🛣️  Routes: {routes[0][0]}")
    
    # Check traffic data
    traffic = fetch_all("SELECT COUNT(*) as count FROM traffic_data")
    print(f"📊 Traffic Data Points: {traffic[0][0]}")
    
    # Get all routes
    all_routes = fetch_all("SELECT route_id, name FROM routes ORDER BY name")
    print("\n📋 Available Routes:")
    for route in all_routes:
        print(f"   - {route[0]}: {route[1]}")
    
    # Latest traffic by route
    latest = fetch_all("""
        SELECT route_id, COUNT(*) as count 
        FROM traffic_data 
        WHERE timestamp >= NOW() - INTERVAL '24 hours'
        GROUP BY route_id
        ORDER BY route_id
    """)
    print("\n📈 Traffic data (last 24h):")
    for row in latest:
        print(f"   {row[0]}: {row[1]} records")
    
    print("\n✅ Database connection successful!\n")

if __name__ == "__main__":
    main()

File: backend/src/test-db.ts
import dotenv from 'dotenv';
import pool, { query } from './utils/database';

// Load environment variables
dotenv.config();

async function testConnection() {
  try {
    console.log('🔍 Testing database connection...\n');
    
    // Test basic connection
    const versionResult = await query('SELECT version()');
    console.log('✅ PostgreSQL Version:', versionResult.rows[0].version.split(',')[0]);
    
    // Check tables
    const tablesResult = await query(
      `SELECT table_name FROM information_schema.tables 
       WHERE table_schema = 'public' 
       ORDER BY table_name`
    );
    console.log('\n📋 Tables found:');
    tablesResult.rows.forEach(row => console.log(`   - ${row.table_name}`));
    
    // Check routes
    const routesResult = await query('SELECT COUNT(*) as count FROM routes');
    console.log(`\n🛣️  Routes: ${routesResult.rows[0].count}`);
    
    // Check traffic data
    const trafficResult = await query('SELECT COUNT(*) as count FROM traffic_data');
    console.log(`📊 Traffic Data Points: ${trafficResult.rows[0].count}`);
    
    // Latest traffic
    const latestResult = await query(
      `SELECT route_id, COUNT(*) as count 
       FROM traffic_data 
       WHERE timestamp >= NOW() - INTERVAL '24 hours'
       GROUP BY route_id`
    );
    console.log('\n📈 Traffic data (last 24h):');
    latestResult.rows.forEach(row => 
      console.log(`   ${row.route_id}: ${row.count} records`)
    );
    
    console.log('\n✅ Database connection successful!\n');
    
  } catch (error) {
    console.error('\n❌ Database connection error:', error);
    process.exit(1);
  } finally {
    await pool.end();
  }
}

testConnection();
